src/expression/comparison/index.ts

```diff
@@ -72,6 +81,28 @@ export abstract class Comparison implements Evaluable {
     if (operator === undefined) {
       throw new Error(`missing operator ${this.operatorSymbol.toString()}`)
     }
-    return [operator, this.left.serialize(options), this.right.serialize(options)]
+    return [
+      operator,
+      this.left.serialize(options),
+      this.right.serialize(options)
+    ]
+  }
+
+  /**
+   * Convert a value to number if possible, otherwise return the value untouched
+   * @param value value to be converted to number
+   */
+  protected toNumber (value: Result): Result {
+    const isValueNumber = isNumber(value)
+    if (isValueNumber) {
+      return value as number
+    } else if (isString(value)) {
+      if (value.match(/^\d+\.\d+$/)) {
+        return parseFloat(value)
+      } else if (value.match(/^0$|^[1-9]\d*$/)) {
+        return parseInt(value)
+      }
+    }
+    return value
```

Comment:
As this function does not depend on state of the class, we should separate it ... just have as a utility function.

Reply:
46e890e